### ISSUE TITLE
Issue regarding model size

### ISSUE THREAD

**Issue by 00ber** [jfisher52/StyleRemix#2]
Hi,
I have a question regarding the huggingface model weights.
I was trying to load some your adapters and play with them but I found that the adapters were very large (~4GB) as in the screenshot below:
![Screenshot 2024-10-30 at 23 31 58](https://github.com/user-attachments/assets/68886dcc-4c67-474e-999c-e369045df852)

Are these weights the full model (Llama-3-8b + LoRA) or is this just the adapter?
I thought these would just be the adapters, but ~4GB seem a little too big for a LoRA, so I just wanted to confirm with you if I'm using the models correctly.

**Comment by shallinan1** [comment]
Hi @00ber , thanks for posting this! These weights should be just the adapters. 4GB does seem large, but I checked and think the weights of Llama-3-8B on its own is ~15GB, so it is smaller relative to the base model. I think we could reduce the size of this checkpoint by changing some of the LoRA hyperparameters during training.

**Comment by 00ber** [comment]
Hi @shallinan1! Thank you for the response! 
Any chance this is unfreezing & training the embedding layer as well?
I ask because when I run the training script I get this warning:

https://github.com/huggingface/peft/blob/b5b902368d2a7f349783a96d9d3c686ea0ca3a05/src/peft/utils/save_and_load.py#L259

In your LoRA training script, I see that you have added a padding token and resized the model embeddings. I suspected that this resize caused the embeddings to be made trainable and thus, saved in the adapter.
I ran a quick experiment by setting padding token to eos_id without adding a new token, and in this case the saved adapter size went down to ~30MB.
